File: week1/Banknotes and coins.py

print("NOTAS:")
for note in notes:
    tot = n[0] // note
    n[0] += -(tot*note)
    print(f"{tot} nota(s) de R$ {note}.00" )


print("MOEDAS:")

##print(n)
if n[0] == 1:
    print(f"1 moeda(s) de R$ 1.00" )
else:
    print(f"0 moeda(s) de R$ 1.00" )
    
for coin in coins:
    tot = n[1] // coin
    n[1] += -(tot*coin)
    
        
    if coin == 5 or coin == 1:
        print(f"{tot} moeda(s) de R$ 0.0{coin}" )
    else:
        print(f"{tot} moeda(s) de R$ 0.{coin}" )

File: week1/banknotes_and_coins.java
package week1;

import java.util.Scanner;

public class banknotes_and_coins {

    public static void main(String[] args) {
        int[] notes = {100,50,20,10,5,2};
        int[] coins = {50, 25, 10, 5 , 1};
        Scanner scan = new Scanner(System.in);

        String n = scan.nextLine();
        String[] ns = n.split("\\.");
        int[] NC = {Integer.parseInt(ns[0]),Integer.parseInt(ns[1])};

        System.out.println("NOTAS:");
        for (int note:
                notes ) {
            int tot = NC[0]/note;
            NC[0] -= tot*note;
            System.out.printf("%d nota(s) de R$ %d.00\n",tot,note);


        }

        System.out.println("MOEDAS:");

        if (NC[0] == 1 ){ System.out.println("1 moeda(s) de R$ 1.00");}
        else { System.out.println("0 moeda(s) de R$ 1.00"); }

        for (int coin :coins) { int tot = NC[1] / coin;
            NC[1] -= tot*coin;


            if (coin == 5 || coin == 1){
                System.out.printf("%d moeda(s) de R$ 0.0%d\n",tot,coin); }
            else{
                System.out.printf("%d moeda(s) de R$ 0.%d\n",tot,coin);  }

        }



    }


}

